cdap-ui/app/cdap/components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/index.js

```diff
@@ -57,7 +57,9 @@ export default class PipelineRuntimeArgsDropdownBtn extends Component {
         // This is to restore it to whatever is the state is in the backend.
         // This will ensure if the user clicks on the "Run" button directly
         // UI will still operate correctly discarding recently entered inputs in the session.
-        fetchAndUpdateRuntimeArgs();
+        if (!this.state.showRunOptions) {
+          fetchAndUpdateRuntimeArgs();
```

Comment:
Per the comment above, `fetchAndUpdateRuntimeArguments` should run when the dialog is closed. If this code change is correct, do you need to update the comment?

Reply:
The code and the comment are still valid. Only now we are updating the runtime arguments when the modal is closed. Previously we were fetching for runtime arguments (macros, and preferences) twice everytime the user opens and closes the modal.